builders/tooling/oracles/phala.md

```diff
@@ -0,0 +1,201 @@
+
+This tutorial will walk through a demo of [interacting with price feeds](#fetch-price-data) enabled by Phat contracts on the demo EVM Appchain. Next, you'll learn how to [deploy price feeds to your own EVM Appchain](#launching-price-feeds-on-your-own-evm-appchain). Please be advised that the steps shown in this tutorial are for demonstration purposes only - it's highly recommended that you [contact the Phala team directly](https://dashboard.phala.network/){target=\_blank} as they can assist you with launching price feeds on your Appchain to ensure the integrity of the deployment process.
+ 
+## A Crash Course on Price Feeds {: #a-crash-course-on-price-feeds }
+
+Before interacting hands-on with price feeds, it's important to understand how price feeds work and how price feed data is accessed by smart contracts. In a standard configuration, each price feed is updated by a decentralized oracle network. Each oracle node is rewarded for publishing the price data to the [aggregator contract](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol){target=\_blank}. 
+
+The aggregator contract receives periodic data updates from the network of oracles and aggregates and stores the data on-chain so that consumers can easily fetch it. However, the information is only updated if a minimum number of responses from oracle nodes are received (during an aggregation round). The end-user can retrieve price feeds with read-only operations via an aggregator interface or via a Consumer interface through the Proxy.
+
+![Price Feed Diagram](/images/builders/tooling/oracles/phala/phala-1.webp)
+
+
+## Fetch Price Data {: #fetch-price-data }
+
+There are several price feeds available on the demo EVM Appchain that you can interact with. The price feeds enabled by Phat Contracts use the same interface as the Chainlink price feeds. The data lives in a series of smart contracts (one per price feed) and can be fetched with the aggregator interface:
+
+???+ code "AggregatorV3Interface.sol"
+
+    ```solidity
+    --8<-- 'code/builders/tooling/oracles/phala/AggregatorV3Interface.sol'
+    ```
+
+As seen above in the interface, there are five functions for fetching data: `decimals`, `description`, `version`, `getRoundData`, and `latestRoundData`. For more information about the `AggregatorV3Interface.sol`, see the [Chainlink API Reference](https://docs.chain.link/data-feeds/api-reference){target=\_blank}.
+
+### Supported Assets {: #supported-assets }
+
+Currently, there are data feed contracts for [the demo EVM Appchain](/builders/tanssi-network/networks/dancebox/demo-evm-appchain/) for the following asset  pairs: 
+
+
+   | Asset & Base Pair |              Aggregator Contract               |
+   |:-----------------:|:----------------------------------------------:|
+   |    AAVE to USD    | {{ networks.dancebox.oracles.phala.aave_usd }} |
```

Comment:
may it would be great to get where they get the pricefeed from even with a link to etherscan etc